orca-front50/src/main/groovy/com/netflix/spinnaker/orca/front50/tasks/SaveServiceAccountTask.java

```diff
@@ -128,9 +131,19 @@ public TaskResult execute(@Nonnull StageExecution stage) {
     }
 
     if (!isUserAuthorized(user, roles)) {
-      // TODO: Push this to the output result so Deck can show it.
       log.warn("User {} is not authorized with all roles for pipeline", user);
-      return TaskResult.ofStatus(ExecutionStatus.TERMINAL);
+      Map<String, Object> operation = new HashMap<>();
```

Comment:
I believe we could just throw an exception here ... `throw new UserException(format("User '%s' is not authorized with all roles for pipeline", user))` and let `orca` map it correctly in the context and set the right `TERMINAL` status.

Reply:
I have change the code, thanks for you advice